source/arm9/libc/nitrofs.c

```diff
@@ -521,6 +521,28 @@ static int nitrofs_stat_file_internal(nitrofs_file_t *f, struct stat *st)
     return 0;
 }
 
+int nitrofs_fat_get_attr(const char *name)
+{
+    if (!nitrofs_local.fnt_offset)
+    {
+        errno = ENODEV;
+        return -1;
+    }
+
+    int32_t res = nitrofs_path_resolve(name);
+    if (res < 0)
+    {
+        errno = ENOENT;
+        return -1;
+    }
+    else if (res >= 0xF000)
+    {
+        return ATTR_DIRECTORY | ATTR_READONLY;
+    }
+
+    return ATTR_READONLY;
```

Comment:
Not sure about "ATTR_ARCHIVE".

Reply:
Windows:

> When set, it indicates that the hosting file has changed since the last backup operation. Windows' file system sets this attribute on any file that has changed.

I think we can safely skip it here.

Reply:
Yeah, that was my thought too, but I wanted an unbiased opinion. Thanks.